fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/CentralPerceptionComponent.java

```diff
@@ -225,63 +225,51 @@ static class MonitoringTrafficObjectIndexProvider extends TrafficObjectIndex {
 
         @Override
         public List<VehicleObject> getVehiclesInRange(PerceptionModel searchRange) {
+            int numberOfVehicles = getNumberOfVehicles();
             try (PerformanceMonitor.Measurement m = monitor.start("search-vehicle")) {
-                m.setProperties(getNumberOfVehicles(), SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
+                m.setProperties(numberOfVehicles, SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
                 return super.getVehiclesInRange(searchRange);
             }
         }
 
         @Override
         public void removeVehicles(Iterable<String> vehiclesToRemove) {
+            int numberOfVehicles = getNumberOfVehicles();
             try (PerformanceMonitor.Measurement m = monitor.start("remove-vehicle")) {
-                m.setProperties(getNumberOfVehicles(), SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
+                m.setProperties(numberOfVehicles, SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
                 super.removeVehicles(vehiclesToRemove);
             }
         }
 
         @Override
         public void updateVehicles(Iterable<VehicleData> vehiclesToUpdate) {
+            int numberOfVehicles = getNumberOfVehicles();
             try (PerformanceMonitor.Measurement m = monitor.start("update-vehicle")) {
-                m.setProperties(getNumberOfVehicles(), SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
+                m.setProperties(numberOfVehicles, SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
                 super.updateVehicles(vehiclesToUpdate);
             }
         }
 
-        @Override
-        public int getNumberOfVehicles() {
-            return super.getNumberOfVehicles();
-        }
-
         @Override
         public List<TrafficLightObject> getTrafficLightsInRange(PerceptionModel perceptionModel) {
             try (PerformanceMonitor.Measurement m = monitor.start("search-traffic-light")) {
-                m.setProperties(getNumberOfTrafficLights(), SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
+                m.setProperties(SimulationKernel.SimulationKernel.getCurrentSimulationTime()).restart();
```

Comment:
I think you can remove this whole line completely. The purpose of the setProperties method when add/remove/update vehicles was, that we could track and evaluate the duration of the calls in relation to the current number of vehicles. Therefore we called `setProperties` and stored the number of vehicles + the current simulation time which was later used to generate charts for our paper. The `restart()` method was required to avoid that the call of `setProperties` was included in the performance measure.

tl;dr just remove this line completely.

Reply:
We discussed this an can leave it for now but will rework this when we overthink the perception indexes.